### DISCUSSION TITLE
Should empty generic and non-generic maybes equal?

### DISCUSSION THREAD

**Discussion by lonix1** [vkhorikov/CSharpFunctionalExtensions#399] [Q&A] [answered]
Theory question:

Should `Maybe<T>.None == Maybe.None`?

Are all empty maybes theoretically identical, or does their container type differentiate them, even though they're empty?

**Comment by vkhorikov** [comment, marked as answer]
Interesting question. I would say they should be treated as different values, similar to how an empty `List<string>` is different from an empty `List<int>`.

**Reply by lonix1** [reply]
Ah yes ... if I remember correctly monads are "containers" and so your explanation makes sense.

**Comment by vkhorikov** [comment]
My previous response applies to cases like `Maybe<string>.None == Maybe<int>.None`.

In your example, the non-generic `Maybe` is just a helpful entry point (a syntax sugar), it doesn't have meaning on its own.
